internal/controller/pipelinerollout_ppnd.go

```diff
@@ -24,11 +24,17 @@ import (
 // - as long as there's no other requirement to pause, set desiredPhase=Running
 // return boolean for whether we can stop the PPND process
 func (r *PipelineRolloutReconciler) processExistingPipelineWithPPND(ctx context.Context, pipelineRollout *apiv1.PipelineRollout,
-	existingPipelineDef, newPipelineDef *kubernetes.GenericObject, pipelineNeedsToUpdate bool) (bool, error) {
+	existingPipelineDef, newPipelineDef *kubernetes.GenericObject) (bool, error) {
 
 	numaLogger := logger.FromContext(ctx)
 
-	needsPaused, err := r.shouldBePaused(ctx, pipelineRollout, existingPipelineDef, newPipelineDef)
+	// this will check for any update required except for "desiredPhase", which PPND
+	pipelineNeedsToUpdate, err := pipelineSpecNeedsUpdating(ctx, existingPipelineDef, newPipelineDef)
+	if err != nil {
+		return false, err
+	}
```

Comment:
we calculate this here instead of using the value that was calculated by the USDE earlier - this eliminates the "lifecycle" field from the comparison (it was still mostly working the other way but the log was a little confusing to read, and we were making an unnecessary change to override `desiredPhase: Running` with ``).